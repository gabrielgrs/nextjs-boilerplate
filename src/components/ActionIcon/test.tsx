import { render, screen, fireEvent } from 'utils/test'
import ActionIcon from './index'

describe('<ActionIcon />', () => {
  it('should match snapshot', () => {
    const { container } = render(<ActionIcon backgroundColor="primary">Action Icon</ActionIcon>)
    expect(container).toMatchSnapshot()
  })

  it('should render right color', () => {
    render(<ActionIcon backgroundColor="primary">Action Icon</ActionIcon>)

    expect(screen.getByText(/action icon/i)).toHaveStyle({
      backgroundColor: '#0bb58d',
    })
  })

  it('should dispatch on click on pressed', () => {
    const onClick = jest.fn()
    render(
      <ActionIcon backgroundColor="primary" onClick={onClick}>
        Action Icon
      </ActionIcon>
    )

    fireEvent.click(screen.getByText(/action icon/i))

    expect(onClick).toHaveBeenCalled()
  })
})
