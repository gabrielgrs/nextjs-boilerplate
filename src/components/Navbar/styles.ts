import styled from 'styled-components'

export const Nav = styled.nav`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 3%;
  width: 100%;
  height: 100%;
`
