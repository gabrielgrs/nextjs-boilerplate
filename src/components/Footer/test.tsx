import { render, screen } from 'utils/test'
import Footer from './index'

describe('<Footer />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Footer />)
    expect(container).toMatchSnapshot()
  })

  it('should render link', () => {
    render(<Footer />)

    expect(screen.getByRole('link', { name: /gabrielgrs/i })).toBeInTheDocument()
  })
})
