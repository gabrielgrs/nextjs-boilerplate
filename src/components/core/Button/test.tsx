import { render, screen } from 'utils/test'
import Button from './index'

describe('<Button />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Button>Button Content</Button>)

    expect(container).toMatchSnapshot()
  })

  it('should render button', () => {
    render(<Button>Button Content</Button>)

    const element = screen.getByRole('button', { name: /Button Content/i })
    expect(element).toBeInTheDocument()
  })

  it('should render a fullWidth version', () => {
    render(<Button fullWidth>Button Content</Button>)

    expect(screen.getByRole('button', { name: /Button Content/i })).toHaveStyle({
      width: '100%',
    })
  })

  it('should render a disabled Button', () => {
    render(<Button disabled>Button Content</Button>)

    expect(screen.getByRole('button', { name: /Button Content/i })).toHaveStyleRule(
      'cursor',
      'not-allowed',
      {
        modifier: ':disabled',
      }
    )
  })

  it('should render Button with variant', () => {
    render(<Button variant="danger">Button Content</Button>)
    const element = screen.getByRole('button', { name: /Button Content/i })
    expect(element).toHaveStyle({
      backgroundColor: '#F5726C',
    })
  })
})
