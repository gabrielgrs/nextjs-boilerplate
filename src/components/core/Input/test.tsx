import userEvent from '@testing-library/user-event'
import { render, screen, fireEvent } from 'utils/test'
import Input from './index'

describe('<Input />', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Input />)

    expect(container).toMatchSnapshot()
  })

  it('Renders with placeholder', () => {
    render(<Input placeholder="Placeholder" />)

    expect(screen.getByPlaceholderText('Placeholder')).toBeInTheDocument()
  })

  it('Changes its value when typing', () => {
    const onChange = jest.fn()
    render(<Input onChange={onChange} placeholder="type something" name="Input" />)
    const input = screen.getByPlaceholderText('type something')

    const typedText = 'new text'
    fireEvent.change(input, { target: { value: typedText } })

    expect(input).toHaveValue(typedText)
  })

  it('Should be disabled', () => {
    const onChange = jest.fn()
    render(<Input disabled onChange={onChange} placeholder="type something" name="Input" />)
    const input = screen.getByPlaceholderText('type something')

    const typedText = 'new text'
    fireEvent.change(input, { target: { value: typedText } })

    expect(input).toBeDisabled()
  })

  it('Renders with error', () => {
    render(<Input invalid placeholder="type something" name="Input" />)

    expect(screen.getByPlaceholderText('type something')).toHaveStyle({
      borderBottom: 'solid 1px #F5726C',
    })
  })

  it('Is accessible by tab', () => {
    render(<Input placeholder="type something" />)

    const input = screen.getByPlaceholderText('type something')
    expect(document.body).toHaveFocus()

    userEvent.tab()
    expect(input).toHaveFocus()
  })
})
