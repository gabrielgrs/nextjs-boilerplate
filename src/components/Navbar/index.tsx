import Image from 'next/image'
import { useRef } from 'react'
import Link from 'next/link'
import useThemeContext from 'hooks/useThemeContext'
import Toggle from '../core/Toggle'
import * as S from './styles'

const Navbar = () => {
  const ref = useRef(null)
  const { isDarkTheme, onSetIsDarkTheme } = useThemeContext()

  return (
    <S.Nav ref={ref}>
      <Link href="/">
        <a>
          <Image src="/assets/logo.svg" height="25px" width="25px" alt="logo" priority />
        </a>
      </Link>

      <Toggle
        name="isDarkMode"
        checked={isDarkTheme}
        onChange={(event) => onSetIsDarkTheme(event.target.checked)}
      />
    </S.Nav>
  )
}

export default Navbar
