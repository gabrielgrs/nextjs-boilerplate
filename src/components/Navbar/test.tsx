import { render } from 'utils/test'
import Navbar from './index'

describe('<Navbar />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Navbar />)
    expect(container).toMatchSnapshot()
  })
})
