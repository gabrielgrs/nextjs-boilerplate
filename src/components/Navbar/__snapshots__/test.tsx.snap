// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match snapshot 1`] = `
.c1 {
  cursor: pointer;
  height: 24px;
  width: 40px;
  background: #282828;
  border-radius: 50px;
  position: relative;
  -webkit-transition: 0.4s all;
  transition: 0.4s all;
}

.c1:after {
  cursor: pointer;
  content: '';
  position: absolute;
  height: 18px;
  width: 18px;
  background: #fefefe;
  border-radius: 50%;
  top: 3px;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  -webkit-transform: translateX(18px);
  -ms-transform: translateX(18px);
  transform: translateX(18px);
}

.c2 {
  cursor: pointer !important;
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 0;
  z-index: 1;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 3%;
  width: 100%;
  height: 100%;
}

<div>
  <nav
    class="c0"
  >
    <a
      href="/"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2725%27%20height=%2725%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="logo"
          data-nimg="intrinsic"
          decoding="async"
          src="/assets/logo.svg"
          srcset="/assets/logo.svg 1x, /assets/logo.svg 2x"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </span>
    </a>
    <div
      class="c1"
      tabindex="0"
    >
      <input
        class="c2"
        name="isDarkMode"
        type="checkbox"
      />
    </div>
  </nav>
</div>
`;
